app/uk/gov/hmrc/eusubsidycompliancefrontend/services/Journey.scala

```diff
@@ -31,60 +30,33 @@ trait FormPage[+T] {
 
 trait Journey {
 
-  private val logger = Logger(getClass)
+  def steps: Array[FormPage[_]]
 
-  protected def steps: List[FormPage[_]]
+  def previous(implicit r: Request[_]): Journey.Uri =
+    if (currentIndex > 0) steps(previousIndex).uri
+    else throw new IllegalStateException("no previous page")
 
-  def formPages: List[FormPage[_]] = steps
-
-  // TODO strip/add the server path prefix instead of endsWith
-  // TODO especially as this may not work for listings
-  def currentIndex(implicit request: Request[_]): Int = {
-    formPages.indexWhere(x => request.uri.endsWith(x.uri))
+  def next(implicit r: Request[_]): Future[Result] = {
+    if (currentIndex < lastIndex) redirectWithSession(steps(nextIndex).uri).toFuture
+    else throw new IllegalStateException("no next page")
   }
 
-  // TODO - should previous and next return the same types?
-  def previous(implicit request: Request[_]): Journey.Uri =
-    formPages.zipWithIndex
-      .find(_._2 == currentIndex - 1)
-      .fold(throw new IllegalStateException("no previous page"))(_._1.uri)
-
-  def next(implicit request: Request[_]): Future[Result] =
-    formPages.zipWithIndex
-      .find(_._2 == currentIndex + 1)
-      .fold(throw new IllegalStateException("no next page")) { x =>
-        Redirect(x._1.uri).withSession(request.session).toFuture
-      }
-
-  def isEmptyFormPage(indexedFormPage: (FormPage[_], Int))(implicit request: Request[_]): Boolean =
```

Comment:
do we not need this? just double checking as it has been removed

Reply:
Its not used anywhere.